src/util/Serializer/SerializeOptional.h

```diff
@@ -10,9 +10,11 @@
 
 // Serialization for `std::optional<T>`
 namespace ad_utility::serialization {
-AD_SERIALIZE_FUNCTION_WITH_CONSTRAINT(
-    (ad_utility::similarToInstantiation<T, std::optional> &&
-     !std::is_trivially_copyable_v<std::decay_t<T>>)) {
+template <typename T>
+CPP_concept similarToOptional =
+    ad_utility::similarToInstantiation<T, std::optional>;
```

Comment:
Maybe this can be done with parentheses and without the explicit `concept`, but that is not important now.